manifests/init.pp

```diff
@@ -27,27 +27,36 @@
 # Copyright 2016 Lukasz Rohde.
 #
 class caddy (
+  String                         $install_path,
+  String                         $caddy_user,
+  String                         $caddy_group,
+  String                         $caddy_log_dir,
+  String                         $caddy_tmp_dir,
+  String                         $caddy_ssl_dir,
+  String                         $caddy_home,
+  Enum['personal', 'commercial'] $caddy_license,
+  Optional[String[1]]            $caddy_account_id,
+  Optional[String[1]]            $caddy_api_key,
+  Enum['on','off']               $caddy_telemetry,
+  String                         $caddy_features,
+  Stdlib::Port                   $caddy_http_port,
+  Stdlib::Port                   $caddy_https_port,
+  Boolean                        $caddy_private_devices,
+  Integer                        $caddy_limit_processes,
+) {
+  case $::architecture {
```

Comment:
`$facts['architecture']` perhaps?

Reply:
There are a lot of facts which require a rewrite.

Reply:
We need a puppet-lint plugin that at least rewrites the built in ones.